vm/src/obj/objlist.rs

```diff
@@ -751,9 +777,9 @@ impl PyList {
         // replace list contents with [] for duration of sort.
         // this prevents keyfunc from messing with the list and makes it easy to
         // check if it tries to append elements to it.
-        let mut elements = self.elements.replace(vec![]);
+        let mut elements = PyList::_replace(self.borrow_elements_mut(), vec![]);
         do_sort(vm, &mut elements, options.key, options.reverse)?;
-        let temp_elements = self.elements.replace(elements);
+        let temp_elements = PyList::_replace(self.borrow_elements_mut(), elements);
```

Comment:
I think if you borrow mutably here, you can just use `std::mem::take()` or `std::mem::replace()`. Something like:
```rust
let mut elements = self.borrow_elements_mut();
let mut temp_elements = std::mem::take(&mut elements);
do_sort(vm, &mut temp_elements, options.key, options.reverse)?;
std::mem::swap(&mut temp_elements, &mut elements);